src/app/components/step-two/step-two.component.ts

```diff
@@ -28,6 +30,31 @@ export class StepTwoComponent implements OnInit {
     return this.formGroup.get(fieldName).value === value;
   }
 
+  onFieldBlur(fieldName: string): void {
+    this.finishedTyping.set(fieldName, true);
+  }
+
+  shouldDisplayField(field: FormField | FormGroupFields): boolean {
```

Comment:
i think i saw this function, if you have duplicate functions please move in parent comp or service and write the once